pandas/core/frame.py

```diff
@@ -3007,6 +3007,12 @@ def _setitem_array(self, key, value):
                 for k1, k2 in zip(key, value.columns):
                     self[k1] = value[k2]
             else:
+                if all(is_hashable(k) for k in key):
```

Comment:
so can you add this in a routine on self.loc

call it 
_ensure_listlike_indexers(key, axis=1)

Reply:
Currently this code always assume that `axis` = 1. Do we need to allow adding multiple rows via assignment too?